src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+
+    if tries <= 0:
+        raise QueryExecutionException("Failed too many times.", query)
+
+    try:
+        context.setQuery(query.load(limit))
+        result = context.query().convert()
+        return result if result else []
+
+    except HTTPError:
+        time.sleep(delay_in_seconds())
+        return execute(query, limit, context, tries - 1)
+
+    except SPARQLExceptions.SPARQLWrapperException as err:
+        raise QueryExecutionException(err.msg, query) from err
+
+    except Exception as err:
+        raise QueryExecutionException(
+            f"{type(err).__name__} - {str(err)}", query
+        ) from err
+
+
+def check_sparql_file(query_file: str) -> Path:
+    """Check meta information of SPARQL query file.
+
+    Args:
+        query_file (str): the file to validate.
+
+    Returns:
+        Path: the validated file.
+    """
+    fpath = Path(query_file)
+
+    if not fpath.is_file():
+        raise argparse.ArgumentTypeError(f"Not a valid file path: {fpath}")
+
+    if fpath.suffix != ".sparql":
+        raise argparse.ArgumentTypeError(f"{fpath} does not have a '.sparql' extension")
+
+    return fpath
+
+
+def check_positive_number(value: str, err_msg: str) -> int:
+    """Ensure 'value' is a positive number.
+
+    Args:
+        value (str): the value to be validated.
+        err_msg (str): used when value fails validation.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated number.
+    """
+    try:
+        number = int(value)
+        if number >= 1:
+            return number
+    except ValueError:
+        pass
+
+    raise argparse.ArgumentTypeError(err_msg)
+
+
+def check_limit(limit: str) -> int:
+    """Validate the 'limit' argument.
+
+    Args:
+        limit (str): the LIMIT to be validated.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated LIMIT
+    """
+    return check_positive_number(
+        limit, "LIMIT must be an integer of value 1 or greater."
+    )
+
+
+def check_timeout(timeout: str) -> int:
+    """Validate the 'timeout' argument.
+
+    Args:
+        timeout (str): the timeout to be validated.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated timeout.
+    """
+    return check_positive_number(
+        timeout, "timeout must be an integer of value 1 or greater."
+    )
+
+
+def main(argv=None) -> int:
+    """The main function.
+
+    Args:
+        argv : If set to None then argparse will use sys.argv as the arguments.
+
+    Returns:
+        int: the exit status - 0: success, any other value - failure.
+    """
+    cli = argparse.ArgumentParser(
+        description=f"run SPARQL queries from the '{PROJECT_ROOT}' project",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    group = cli.add_mutually_exclusive_group(required=True)
+
+    group.add_argument(
+        "-c",
+        "--changed",
+        action="store_true",
```

Comment:
This is so cool. Like really I'm so happy to see how this can be done! Really fascinating to see this in action!